MapboxMobileEvents/MMEAPIClient.m

```diff
@@ -66,21 +77,23 @@ - (void)postEvents:(NSArray *)events completionHandler:(nullable void (^)(NSErro
     
     for (NSArray *batch in eventBatches) {
         NSURLRequest *request = [self requestForEvents:batch];
-        [self.sessionWrapper processRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
-            NSError *statusError = nil;
-            if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
-                NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
-                statusError = [self statusErrorFromRequest:request andHTTPResponse:httpResponse];
-            } else {
-                statusError = [self unexpectedResponseErrorfromRequest:request andResponse:response];
-            }
-            if (completionHandler) {
-                error = error ?: statusError;
-                completionHandler(error);
-              
-                [self.metricsManager updateMetricsFromEvents:events request:request error:error];
-            }
-        }];
+        if (request) {
+            [self.sessionWrapper processRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+                NSError *statusError = nil;
+                if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
+                    NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+                    statusError = [self statusErrorFromRequest:request andHTTPResponse:httpResponse];
+                } else {
+                    statusError = [self unexpectedResponseErrorfromRequest:request andResponse:response];
+                }
+                if (completionHandler) {
+                    error = error ?: statusError;
+                    completionHandler(error);
+
+                    [self.metricsManager updateMetricsFromEvents:events request:request error:error];
```

Comment:
Looks like this should be outside the `if (completionHandler)` block?
Also should `updateMetricsFromEvents:request:` be passed `batch` instead of `events`?

Reply:
Good catch. That should be the case. Passing in the batch shouldn't matter for our current setup but it certainly will for deferred telemetry.